src/sinol_make/util.py

```diff
@@ -90,33 +98,39 @@ def install_oiejq():
         with open(oiejq_path, 'wb') as oiejq_file:
             oiejq_file.write(request.content)
 
-        def strip(tar):
-            l = len('oiejq/')
-            for member in tar.getmembers():
-                member.name = member.name[l:]
-                yield member
-
         with tarfile.open(oiejq_path) as tar:
-            tar.extractall(path=os.path.expanduser('~/.local/bin'), members=strip(tar))
-        os.rename(os.path.expanduser('~/.local/bin/oiejq.sh'), os.path.expanduser('~/.local/bin/oiejq'))
+            tar.extractall(path=tmpdir)
+
+        # If the user has a directory named `oiejq` in `~/.local/bin`,
+        # we rename the `oiejq.sh` executable to `oiejq_sinol-make`.
+        # Otherwise, we rename it to `oiejq`.
```

Comment:
Maybe it's better to check whether `oiejq` can be run, and otherwise move it? That is, maybe we don't have to check whether it is a directory or not?

Reply:
If there exists a file named `oiejq` in `.local/bin/` and it is executable, then it will be used.
I don't think that moving user's files is good

Reply:
Then I would prefer to make an error with an understandable error message.

Reply:
Thanks to that, the user would do whatever he wants with his files, and we won't have to lengthen the code because of edge-cases.

Reply:
Now the code fails before installing if `~/.local/bin/oiejq` already exists